Lib/Recycled.hpp

```diff
@@ -211,8 +211,8 @@ class Recycled
   T* operator->() { return &self(); }
 
   auto size() const { return self().size(); }
-  template<class Idx> auto operator[](Idx idx) const { return self()[idx]; }
-  template<class Idx> auto operator[](Idx idx)       { return self()[idx]; }
+  template<class Idx> decltype(auto) operator[](Idx idx) const { return self()[idx]; }
+  template<class Idx> decltype(auto) operator[](Idx idx)       { return self()[idx]; }
```

Comment:
`decltype(auto)` rather than `auto` here because you want to allow `T`, `T &`, `T &&` and cv-qualified friends?

Reply:
Yes exactly. For example in a `Recycled<Stack<int>>` we want to be able to get an `int&` when we index into it, and not just an `int`.